tests/test_all_rules.py

```diff
@@ -640,6 +640,40 @@ def test_deprecated_rules_modified(self):
         if result:
             self.fail(f"Deprecated rules {result} has been modified")
 
+    def test_rule_change_has_updated_date(self):
+        """Test to ensure modified rules have updated_date field updated."""
+
+        rules_path = get_path("rules")
+        rules_bbr_path = get_path("rules_building_block")
+
+        # Use git diff to check if the file(s) has been modified in rules/ rules_build_block/ directory
+        # For now this checks even rules/_deprecated any modification there will fail
+        # the test case "test_deprecated_rules_modified", which means an ignore directory
+        # is not required as there is a specific test for deprecated rules
+
+        detection_rules_git = make_git()
+        result = detection_rules_git("diff", "--diff-filter=M", "origin/main", "--name-only",
+                                     rules_path, rules_bbr_path)
+
+        # If the output is not empty, then file(s) have changed in the directory(s)
+        if result:
+            rules = result.splitlines()
+            rc = RuleCollection()
+            failed_rules = []
```

Comment:
Just something to consider: At this point, since we're using the [BaseRuleTest](https://github.com/elastic/detection-rules/pull/3764/files#diff-143495f0794a3e8b322e5cbf9a4735acab2213fd6d0f678149b78d322976507bR552) class, you really dont even have create a new `RuleCollection()`. The rules are already loaded in `self.production_rules`, `self.bbr`, and `self.deprecated_rules`. 

Instead of loading, you could just compare path from the `result` with the `rule.path`. Not how this will improve / degrade the speed of this unit test (iterating over all rules vs loading 'ideally' a smaller set of rules).

Reply:
Couple of things why the Rule collection was used in the older implementation 
- We had to fetch the updated_date from the rule to compare to the last_commit_date [ Which we have scrapped]
- When we moved to a complete git based implementation of checking for modified files and relaying on git diff information, we did not want any rule detail at all, but we kept the rule collection to use for fetching rule_str for error messages. 
- I don't think its worth the computation to iterate over any rules that are loaded (or even a new collection) just to fetch the Rule id and Rule name for erroring, instead we can use the path for error message below like 
```
AssertionError: 
E                               The following rules in the below path(s) have been modified but updated_date has not been changed 
E               
E                               rules/linux/persistence_apt_package_manager_file_creation.toml
E               rules/promotions/endgame_ransomware_prevented.toml
E               rules_building_block/discovery_files_dir_systeminfo_via_cmd.toml```

- I am removing the code block for rule loading with this